python/spark_sklearn/base_search.py

```diff
@@ -0,0 +1,172 @@
+"""
+Base class for parallelizing CV search jobs in scikit-learn
+"""
+
+from collections import defaultdict, Sized
+from functools import partial
+from itertools import islice
+from random import randint
+import numpy as np
+from scipy.stats import rankdata
+
+from sklearn.base import is_classifier, clone
+from sklearn.metrics.scorer import check_scoring
+from sklearn.model_selection import check_cv
+from sklearn.model_selection._validation import _fit_and_score
+from sklearn.model_selection._search import BaseSearchCV
+from sklearn.utils.fixes import MaskedArray
+from sklearn.utils.validation import indexable
+
+
+class SparkBaseSearchCV(BaseSearchCV):
+
+    def __init__(self, estimator, scoring=None, fit_params=None, n_jobs=1, iid=True, refit=True,
+                 cv=None, verbose=0, pre_dispatch='2*n_jobs', error_score='raise',
+                 return_train_score=True):
+        super(SparkBaseSearchCV, self).__init__(
+            estimator=estimator, scoring=scoring, fit_params=fit_params, n_jobs=n_jobs, iid=iid,
+            refit=refit, cv=cv, verbose=verbose, pre_dispatch=pre_dispatch, error_score=error_score,
+            return_train_score=return_train_score)
+
+    def _fit(self, X, y, groups, parameter_iterable):
+
+        estimator = self.estimator
+        cv = check_cv(self.cv, y, classifier=is_classifier(estimator))
+
+        if hasattr(cv, 'random_state'):
+            if not cv.random_state:
+                cv.random_state = randint(1000, 9999)
+
+        self.scorer_ = check_scoring(self.estimator, scoring=self.scoring)
```

Comment:
I am always confused by `_scorer` vs `scorer_`. No preference really.

Reply:
I don't know enough to have an opinion; I'll leave it unless there's a reason to change it here.

Reply:
leading `_` on a variable/method is mostly for internal use in python.  Sort of like a private class member. You won't be able to import a variable with leading `_`. 
A trailing `_` is used when you want to use a name which is a python keywords.